### DISCUSSION TITLE
v2.0.0-beta.42

### DISCUSSION THREAD

**Discussion by 0xJacky** [0xJacky/nginx-ui#784] [Announcements]
# Notice
We're excited to announce that we've been hard at work developing the next generation of the Nginx UI (v2). To keep up-to-date with our progress, please visit https://github.com/users/0xJacky/projects/1.

Please feel free to send any feedback, suggestions, or bug reports on Github issues.

Thank you to all of the beta testers as we approach rc.1.

# Change vs v2.0.0-beta.41
1. Added a sponsor button on the About page.
2. Added support for Azure API type of OpenAI #475 .
3. Added external URL for casdoor #603.
4. Fixed maybe fail to sync config to remote node #775.
5. Added ICP number and Public Security number on the footer of the login page for users in Mainland China #780 .

# Breaking Changes:
1. Starting from v2.0.0-beta.37, we introduced a new version of the structure of the settings file(v2). When you upgrade to the latest version, it will automatically perform the migration process. Please make sure that the Nginx UI in your server has permission to modify the config file itself. Please check the documentation for more details: https://nginxui.com/.
17. Starting from v2.0.0-beta.36, you must specify the directories where the logs are stored. Only logs within these directories can be viewed online. Check this documentation for more information: https://nginxui.com/guide/config-nginx.html#logdirwhitelist.
18. The `CasdoorSettings.Certificate` setting has been deprecated and replaced by `CasdoorSettings.CertificatePath`, which refers to the path of the certificate file.
19. About PR #445, if you are using nginx-ui docker container and encounter this issue, you have to remove the trailing `.conf` manually. Diff: https://github.com/0xJacky/nginx-ui/pull/445/files
20. The `nginx_log` section in the configuration file has been renamed to `nginx`.
`NginxConfigDir` and `NginxPIDPath` in the `server` section have been renamed to `ConfigDir` and `PIDPath` respectively, and both have been moved into the `nginx` section.
21. The `ReloadCmd` and `RestartCmd` have been added in the `nginx` section to configure custom commands for reloading/restarting Nginx.
22. In versions above v2.0.0-beta.23 of Nginx UI Docker containers, `/var/logs/nginx/access.log` and `/var/logs/nginx/error.log` are no longer soft links to `/dev/stdout` and `/dev/stderr`, but regular files. If you need this functionality, please add `access_log /dev/stdout` and `error_log /dev/stderr` in your `nginx.conf`. For users upgrading from older versions who do not need this feature, check if your `nginx.conf` contains `access_log /var/logs/nginx/access.local.log` and `error_log /var/logs/nginx/error.local.log`. If so, please manually remove them to avoid duplicate log entries. This change allows you to map `/var/logs/nginx` to a directory on the host machine to achieve persistent storage of log files.
<details> 

<summary>What's new:</summary>

# What's new
1. We're proud to have registered the [nginxui.com](https://nginxui.com) domain 🎉.
23. We have launched a new documentation site, please check [nginxui.com](https://nginxui.com). 
24. Node management functionality has been introduced (#70).
25. Duplicate site function now can duplicate config to a remote server.
26. French translations have been added, thanks to the contributions of @skrilax91.
27. Added border-radius to Ace Editor and terminal, we hope you will like this change :).
28.  You can now deploy site configuration to a remote server.
29. Spanish translations have been added, thanks to the contributions of @kcholoren.
30. Russian translations have been added, thanks to the contributions of @iAHTOH.
31. The dashboard now includes a system monitor for nodes.
32. Use jsoniter for improving gin performance.
33. Added custom reload/restart command for nginx. #140
34. Refacted translation module #135 
35. Added HTTP host config #157 by @quanbisen 
36. Add setting for test nginx config command #191 
37. Added casdoor sso support, thanks to the contributions of @Jraaay 
38. Refactor the project directory structure.
39. Use tailwind css.
40. Use eslint to check the code style of frontend.
41. Use vue-tsc to check the types for frontend.
42.  Introducing a new management page for certificates, users can now view the latest renewal logs of certain certificates.
43. Add a manually renew button in the certificate modify page for the managed certificate (auto cert).
44. Add notifications for renewing certificates. #192 
45. Automatically detect the path of access logs and error logs.
46. Introducing the feature of issuing wildcard certificates on the Manage Certificate page. #190 
47. Add some Nginx Config Templates #233, thanks to the contributions of @sanvu88 . 
48. Support different key types for obtaining certificates, including RSA2048, RAS3072, RSA4096, EC256 and EC384 #264.
49. Introducing logroate cron task for docker users #255.
50. A system default acme user will be registered when nginx ui starts. https://github.com/0xJacky/nginx-ui/issues/237
51. Add Acme user management page. https://github.com/0xJacky/nginx-ui/issues/319
52. Persist the Acme user to the database.
53. Add a setting of renewal interval of certificates (7-21d). https://github.com/0xJacky/nginx-ui/issues/343
54. Add an option to select an Acme user when obtaining a certificate. https://github.com/0xJacky/nginx-ui/issues/319 https://github.com/0xJacky/nginx-ui/issues/16
55. SSL management supports different types of certificates of the same domain name. https://github.com/0xJacky/nginx-ui/issues/309, https://github.com/0xJacky/nginx-ui/issues/288
56. Certificates obtained within 21 days are renewed by using the renewal API. For compatibility, the challenge resolver is still set up when renewing. https://github.com/0xJacky/nginx-ui/issues/319
57. Add trash for soft deleted items in CURD pages.
58. Use s6-overlay to manage the services in docker #308 .
59. Users can use settings or environment variables to skip installation #357 .
60. Users can use environment variables to predefine an admin user  (only available in skip installation mode) #214 .
61. Users can customize the name of local environment #313 .
62. Users can use the settings file to predefine nodes #169 .
63. Hot reload cluster node settings from settings file #169.
64. Introducing synchronize certificates to remote server #293, #363, #411.
65. Enabled search box for Ace code editor #412.
66. Add a white list of IPs, only those in the list can access the Nginx UI. 
67. Add login failed ban IP list. 
68. Add 2FA authorization for login #344, #419.
69. Add 2FA authorization for Web Terminal #394 .
70. Hide some parts of Jwt Secret and Node Secret on the settings page.
71. The selector of the changed certificate now supports selecting multiple certificates #437 .
72. Certificate(s) Info now supports display info of multiple certificates.
73. The info on changed certificates in the server tab will be shown. #386
74. Better breadcrumbs #258.
75. Add OCSP Must Staple options in AutoCert #292.
76. Add LEGO_DISABLE_CNAME_SUPPORT options in AutoCert #407.
77. Added the capability to create new configuration files online.
78. Introduced a feature to rename configuration files.
79. Better breadcrumbs for configurations list and editor.
80. Add filename filter for configurations list.
81. Added the capability to create a new folder in the configurations list.
82. Added the capability to rename folders or files in the configurations list.
83. Implemented deployment of configuration files to remote nodes.
84. Update translations of Korean, thanks for @jkh0kr .
85. Added Portuguese translations—special thanks to @mksarifo.
86. Support Webauthn passkey as a login and 2FA method.
87. Added site category. #657 
88. Added status filter for site list. #633 
89. Added selection box for Nginx Config Template.
90. Added scheme and host fields for the reverse proxy template. #608

# Fixes:
1. We've upgraded gonginx to address the configuration parsing issue #73.
16. Fixed an issue where it was not possible to update the settings value to an empty string.
17. Resolved a location-saving issue on the domain addition page #116.
18. Site enabling/disabling is now aborted only when the log level is higher than warning.
19. Fixed environment indicator display overlay and data obtaining issues #152 by @quanbisen 
20. Fixed env indicator background in dark mode.
21. Terminal process not killed #181 
22. Added `gpt-4-1106-preview` model in ChatGPT settings.
23. Fixed change language cause unexpected redirect #189 
24. Fixed English grammar error in button text on site management page. #200 
25. Fixed logs mnual link error. #201 
26. Revert enable site if you trying to enable an incorrect config of the site. #199 
27. Upgrade ant design vue version from v3 to v4, thanks @funcfang
28. Fixed FooterToolBar content was covered by Monica, a Chrome Extension. #195 
29. Delete server confirm modal not destroy #206 
30. clientMaxBodySize name is wrong. #216 
31. Fix errors may occur when enabling TLS. #230 
32. Introducing stream management pages. #166 
33. Introducing upstream editor. #138 
34. Introducing upstream health check (supports TCP and Unix Socket).
35. Fix a bug that may cause nginx-ui to take 100% CPU if the log file is not a regular file.
36. Fix a bug of the loading indicator constantly spinning in the table of configurations management page, if the directory has no file.

# Enhances:
1. The recovery function now can print call stack information.
13. Added file info to config editor.
14. Improved zh_TW locale. #148 
15. Enhance the error handler for execShell.
16. Enhance nginx control and status acquisition. #222 
17. Remove the validated rules for DNS Credentials

# Docs:
1. Traditional Chinese documentation has been added, thanks to @Hintay.
10. We've added documentation for the syntax of the configuration template.
12. We have updated the documentation for the Nginx proxy example.
18. Updated deprecated TLS directives #183 

# Others:
1. Core upgrades will be performed as a dry run in demo mode.
9. Updated zh_CN translations.
10. Changed Token to NodeSecret #152 
13. Updated configurations of DNS challenges provider.
</details> 

<hr /><em>This discussion was created from the release <a href='https://github.com/0xJacky/nginx-ui/releases/tag/v2.0.0-beta.42'>v2.0.0-beta.42</a>.</em>

**Comment by leic4u** [comment]
I supposed this release would be the rc release, it seems I am wrong. 😂

**Reply by 0xJacky** [reply]
Haha, we still need some time.
